packages/firebase_messaging/android/src/main/java/io/flutter/plugins/firebasemessaging/FlutterFirebaseMessagingService.java

```diff
@@ -129,9 +129,18 @@ public void onNewToken(String token) {
     LocalBroadcastManager.getInstance(this).sendBroadcast(intent);
   }
 
+  /**
+   * Setup the background isolate that would allow background messages to be handled
+   * on the Dart side. Called either by the plugin when the app is starting up or when
+   * the app receives a message while it is inactive.
+   *
+   * @param context Registrar or FirebaseMessagingService context.
+   * @param callbackHandle Handle used to retrieve the Dart function that sets up
+   *                       background handling on the dart side.
+   */
   public static void startBackgroundIsolate(Context context, long callbackHandle) {
     FlutterMain.ensureInitializationComplete(context, null);
-    String mAppBundlePath = FlutterMain.findAppBundlePath(context);
+    String appBundlePath = FlutterMain.findAppBundlePath();
```

Comment:
error: method findAppBundlePath in class FlutterMain cannot be applied to given types;

` String appBundlePath = FlutterMain.findAppBundlePath();`

required: Context
  found: no arguments
  reason: actual and formal argument lists differ in length

Not sure why the context was removed on your latest commit but it creates a compile time error with the removed context

Reply:
> error: method findAppBundlePath in class FlutterMain cannot be applied to given types;
> 
> ` String appBundlePath = FlutterMain.findAppBundlePath();`
> 
> required: Context
> found: no arguments
> reason: actual and formal argument lists differ in length
> 
> Not sure why the context was removed on your latest commit but it creates a compile time error with the removed context

how to fix above error in background? E/FlutterFcmService: Fatal: failed to find callback 
any advice?

Reply:
Hmmm, this may be a [method that is deprecated](https://github.com/flutter/engine/commit/4c6948a88af62b5c79e93073cf974e9130e84abe#diff-599e1d64442183ead768757cca6805c3R323) on the master channel. I'll revert to the previous method since that has not come to the stable channel yet.

Thanks for flagging this.

Reply:
> Hmmm, this may be a [method that is deprecated](https://github.com/flutter/engine/commit/4c6948a88af62b5c79e93073cf974e9130e84abe#diff-599e1d64442183ead768757cca6805c3R323) on the master channel. I'll revert to the previous method since that has not come to the stable channel yet.
> 
> Thanks for flagging this.

now.. toRawHandle() error with red screen after running example app...

Reply:
@pafnat where are you getting that error, do you have a stack trace?

Reply:
> @pafnat where are you getting that error, do you have a stack trace?

suppose this error on 122 or 123 lines of firebase_messaging.dart

Reply:
> > Hmmm, this may be a [method that is deprecated](https://github.com/flutter/engine/commit/4c6948a88af62b5c79e93073cf974e9130e84abe#diff-599e1d64442183ead768757cca6805c3R323) on the master channel. I'll revert to the previous method since that has not come to the stable channel yet.
> > Thanks for flagging this.
> 
> now.. getting this:
> NoSuchMethodError: The method 'toRawHandle' was called on null.

I'm getting the same error. At least its now on the dart side. Everything on the native seems alright. Looks like somethings wrong with the incoming MessageHandler onBackgroundMessage from our custom implementation on our dart side in the fcm.configure(). Returning null on getCallbackHandle. Lines 117 and 118 on the dart. Subsequently hitting a runtime at line 123.

Reply:
> > > Hmmm, this may be a [method that is deprecated](https://github.com/flutter/engine/commit/4c6948a88af62b5c79e93073cf974e9130e84abe#diff-599e1d64442183ead768757cca6805c3R323) on the master channel. I'll revert to the previous method since that has not come to the stable channel yet.
> > > Thanks for flagging this.
> > 
> > 
> > now.. getting this:
> > NoSuchMethodError: The method 'toRawHandle' was called on null.
> 
> I'm getting the same error. At least its now on the dart side. Everything on the native seems alright. Looks like somethings wrong with the incoming MessageHandler onBackgroundMessage from our custom implementation on our dart side in the fcm.configure(). Returning null on getCallbackHandle. Lines 117 and 118 on the dart. Subsequently hitting a runtime at line 123.

@kroikie @XSiyabonga
Does it mean, that there are no working solution for this issue right now? i was going to rapidly make an app (with learning Flutter in parallel) and this issue ic critical for me. And if anyone needs make some tasks in background (when your app closed or minimized), - for example catch push notification, or listening for incoming SMS to parse.. better to choice native code for Android platform then Flutter?

Reply:
Finished checking #1898 pull request,
It seems there (with class FlutterBackgroundMessagesReceiver extends BroadcastReceiver) we are able to catch notification after we close the app.
Is there way to do the same on Dart side here?

Reply:
@XSiyabonga The README has an example of what `onBackgroundMessage` handler would look like.

@pafnat what solution are you looking for. This PR gives the ability to receive and handle FCM messages on Android when the app is in the background. Could you give more context on what is not working for you?

Reply:
> @XSiyabonga The README has an example of what `onBackgroundMessage` handler would look like.
> 
> @pafnat what solution are you looking for. This PR gives the ability to receive and handle FCM messages on Android when the app is in the background. Could you give more context on what is not working for you?

I was learning Flutter and there is an interesting task for automation my gate at home. The app should parse incoming SMS (or better parse incoming Push notification) while running always in background (or closed completely). Then the phone should make a call automatically on concrete phone number. Seems it is very easy task. I can make it natively without Flutter on Android but i want to measure how to achieve the same with Flutter and check both sources.
So, for Flutter it would be perfect if all loic would be on Dart side for example:
1. Parse incoming push from firebase (if app runnin or closed)
2. Check secret code inside push
3. If code is right, then make a call

2 and 3 i have already finished in Dart (there is a plugin exists for calling for Flutter)
But now i want to do the 1 with Dart too(without changing the logic on Java side)

Reply:
@pafnat `toRawHandle` is called only in two places. Lines 121 and 122 of `firebase_messaging.dart`. If that is being called on null then either the backgroundSetupHandle or backgroundMessageHandle is null. Since the plugin already defines `backgroundSetupHandle` that is unlikely to be null, so maybe the `backgroundMessageHandle` is null (a stack trace would confirm).

Could you verify that you are setting the `onBackgroundMessage` parameter of the `configure` method?

Reply:
> `onBackgroundMessage`

sure, i have used onBackgroundMessage  as described at README. still  the same error as above.

Reply:
@pafnat, could you confirm exactly where you are getting that error. If you provide a stack trace it would help track down the problem.

Reply:
> @pafnat, could you confirm exactly where you are getting that error. If you provide a stack trace it would help track down the problem.

I will try but in 1-2 days.
Anyway i have made my task with #1898 PR. Can confirm, as temporaly solution to achieve my task.
Even if i close completely Flutter app, firebase messages are delivered and after that, app make outgoing phone call automatically. Thats what i needed.
Can you confirm that in that #1900 PR should be the same possible, or, only true in case the app is not closed by user?

PS. In Android Q you have to manually open developer options and enable start activities from bachrounf service toggle. By default it will be failed. 
And it means, that this #1900 PR will not work on Android 10....

Reply:
@pafnat, I'm glad you have a solution. This PR covers all use cases of #1898 so if that one worked this will work as well.

The FirebaseMessagingService is a special service that is managed by Google Play Services so the Q changes will not affect the ability of messages to be handled in the background.

Reply:
> @pafnat, I'm glad you have a solution. This PR covers all use cases of #1898 so if that one worked this will work as well.
> 
> The FirebaseMessagingService is a special service that is managed by Google Play Services so the Q changes will not affect the ability of messages to be handled in the background.

The problem that in Android Q dp5 it occurred. To quick fiz tou have to enable toggle in developers options to execute code from backround. 
In Android < 10, it works out of box without going tondevelopers options.